cdap-docs/tools/doc-change.py

```diff
@@ -0,0 +1,206 @@
+#  Filename is accepted as the first command line arg and the current version is accepted as the second arg.
+def append_links(filepath, version, ghpages=False, ending='', current=None):
+    # Get file content
+    file_object = open(filepath, 'r')
+    file_string = file_object.read()
+    file_object.close()
+    dirty = False
+
+    # Modify File String
+    
+    # Look for an empty title
+    empty_title_tags = '<title>%s</title>'
+    empty_title_tags_index = file_string.lower().find(empty_title_tags % '')
+    if empty_title_tags_index != -1:
+        title_version = current if version == 'current' else version
+        empty_title_tags_end_index = empty_title_tags_index + len(empty_title_tags % '')
+        file_string = file_string[:empty_title_tags_index] + empty_title_tags % "CDAP %s" % title_version + file_string[empty_title_tags_end_index:]
+        dirty = True
+        
+    # Set canonical headers on GitHub pages (indicated by --ghpages option) and current pages
+    if (ghpages and (version == current or version == 'current')) or (version == 'current'):
+        domain = 'docs.cask.co'
+        if ghpages and version == 'current':
+            replacement_version = version
+        else:
+            replacement_version = current
+            
+        canonical_link_tag = '\n\n    <link rel="canonical" href="http://%s/cdap/%s">\n' % (domain, replacement_version + ending)
+        close_title_tag = '</title>'
+        close_title_tag_index = file_string.lower().find(close_title_tag)
+        if close_title_tag_index != -1:
+            close_title_tag_index = close_title_tag_index + len(close_title_tag)
+            file_string = file_string[:close_title_tag_index] + canonical_link_tag + file_string[close_title_tag_index:]
+            dirty = True
+            
+    # Set robots meta-tag no-index no-follow on GitHub pages non-current pages and Docs.cask.co non-current_numbered_version
+    if version != 'current' and version != current:
+        meta_tag = '\n    <meta name=\"robots\" content=\"noindex, nofollow\">\n'
+        open_head_tag = '<head>'
+        open_head_tag_index = file_string.lower().find(open_head_tag)
+        if open_head_tag_index != -1:
+            open_head_tag_index = open_head_tag_index + len(open_head_tag)
+            file_string = file_string[:open_head_tag_index] + meta_tag + file_string[open_head_tag_index:]
+            dirty = True
+
+    if dirty:
+        # Write new content to file
+        file_object = open(filepath, 'w')
+        file_object.write(file_string)
+        file_object.close()
+
+def convert_docs(doc_set_path, ghpages=False, current=None):
+    doc_set = os.path.basename(doc_set_path)
+    print "Converting doc set '%s' (%s)" % (doc_set, doc_set_path)
+    html_count = 0
+    for root, dirs, files in os.walk(doc_set_path):
+        for f in files:
+            if f.endswith('.html'):
+                html_count += 1
```

Comment:
I think that you should assign the matching files to an array here, which you will loop through below. There's no reason to reset `html_count` and reuse it as you're performing the same conditional check below.